### PR TITLE
add history processing lambda

### PR DESCRIPTION
**Pull request by ashley-taylor**

### CODE REVIEW

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		builder.dynamodb(toV2(record.getDynamodb()));
+		builder.eventID(record.getEventID());
+		builder.eventName(record.getEventName());
+		builder.eventSource(record.getEventSource());
+		builder.eventVersion(record.getEventVersion());
+		builder.userIdentity(toV2(record.getUserIdentity()));
+		return builder.build();
+	}
+
+	public static Identity toV2(com.amazonaws.services.dynamodbv2.model.Identity userIdentity) {
```
Comment (Aliics) [inline graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java:30]: Don't need to use the fully qualified class name.

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		builder.oldImage(toV2(dynamodb.getOldImage()));
+		builder.sequenceNumber(dynamodb.getSequenceNumber());
+		builder.sizeBytes(dynamodb.getSizeBytes());
+		builder.streamViewType(dynamodb.getStreamViewType());
+
+		return builder.build();
+	}
+
+	public static Map<String, AttributeValue> toV2(
+			Map<String, com.amazonaws.services.dynamodbv2.model.AttributeValue> attribute) {
```
Comment (Aliics) [inline graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java:52]: Don't need to use the fully-qualified class name.

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/HistoryLambda.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	public abstract String getTableName();
+
+	public abstract DynamoDbAsyncClient getClient();
+
+	@Override
+	public Void handleRequest(DynamodbEvent input, Context context) {
+		var records = input.getRecords().stream().map(DynamoUtil::toV2);
+		process(records);
+		return null;
+	}
```
Comment (Aliics) [inline graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/HistoryLambda.java:36]: Would be nice to have the output for potential debugging purposes.

Would require doing a little bit of gross CF coercing though.

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java (hunk 1 above)
Comment (ashley-taylor) [reply graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java:30]: one on the left is software.amazon different classes

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java (hunk 2 above)
Comment (ashley-taylor) [reply graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java:52]: one on the left is software.amazon different classes

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/HistoryLambda.java (hunk 3 above)
Comment (ashley-taylor) [reply graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/HistoryLambda.java:36]: risk of outputting is the cloud watch cost will greatly exceed the lambda processing charge.

File: graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java (hunk 2 above)
Comment (Aliics) [reply graphql-database-dynmodb-history-lambda/src/main/java/com/fleetpin/graphql/database/dynamo/history/lambda/DynamoUtil.java:52]: ah yep, didn't notice that, should've been obvious from the behaviour of the method oops.

**Pull request merged**